### PR TITLE
Enhacements.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️

### CODE REVIEW

File: bin/yii [hunk 1]
```diff
@@ -1,12 +1,18 @@
 #!/usr/bin/env php
 <?php
 
+declare(strict_types=1);
+
+defined('YII_ENV') or define('YII_ENV', 'DEV'); // PROD, DEV
```
Comment (samdark) [inline bin/yii:6]: How would you redefine it?

File: src/Command/Serve.php [hunk 2]
```diff
@@ -68,6 +71,10 @@ protected function execute(InputInterface $input, OutputInterface $output)
         }
         $output->writeLn('Quit the server with CTRL-C or COMMAND-C.');
 
+        if ($env === 'test') {
```
Comment (samdark) [inline src/Command/Serve.php:74]: What's the goal here?

Comment (terabytesoftw) [reply src/Command/Serve.php:74]: Do the unit tests, to test the command.

File: bin/yii (hunk 1 above)
Comment (terabytesoftw) [reply bin/yii:6]: The problem is composer-plug-in adds env variables, or constants to parameters, when variables and constants must be available to the environment.

Comment (samdark) [reply bin/yii:6]: `getenv()`

File: bin/yii [hunk 3]
```diff
@@ -37,12 +42,23 @@ use Yiisoft\Yii\Console\Output\ConsoleBufferedOutput;
...
 
-    $params = require Builder::path('params');
+    // Don't do it in production, assembling takes it's time
+    if (getenv('env') === 'dev') {
+        Builder::rebuild();
+    }
+
+    $startTime = microtime(true);
+
+    $container = new Container(
```
Comment (samdark) [inline bin/yii:55]: It makes it impossible to replace container with another implementation.

Comment (terabytesoftw) [reply bin/yii:55]: It's yii-console, so it makes sense to me to use the yii-di container.

File: config/providers-console.php [hunk 4]
```diff
@@ -1,3 +1,18 @@
 <?php
 
-return [];
+declare(strict_types=1);
+
+use Yiisoft\Yii\Console\Provider\ApplicationProvider;
+use Yiisoft\Yii\Console\Provider\EventDispatcherProvider;
+
+return [
+    'yiisoft/event-dispatcher/eventdispatcher' => EventDispatcherProvider::class,
```
Comment (rustamwin) [inline config/providers-console.php:9]: Does the suffix matter?

Comment (terabytesoftw) [reply config/providers-console.php]: no, but fixed.

File: bin/yii [hunk 5]
```diff
@@ -37,12 +42,23 @@ use Yiisoft\Yii\Console\Output\ConsoleBufferedOutput;
         fwrite(STDERR, $message);
         exit(1);
     }
+
     require_once $autoloadPath;
 
-    $params = require Builder::path('params');
+    // Don't do it in production, assembling takes it's time
+    if (getenv('env') === 'dev') {
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply bin/yii]: done.

**Comment by samdark** [comment]
👍

File: tests/data/index.php [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+declare(strict_types=1);
+
+use Psr\Container\ContainerInterface;
+use Yiisoft\Composer\Config\Builder;
+use Yiisoft\Di\Container;
+use Yiisoft\Http\Method;
+use Yiisoft\Yii\Web\Application;
+use Yiisoft\Yii\Web\Config\EventConfigurator;
+use Yiisoft\Yii\Web\SapiEmitter;
+use Yiisoft\Yii\Web\ServerRequestFactory;
```
Comment (roxblnfk) [inline tests/data/index.php:12]: Why do you need this file with all thise dependencies?

**Pull request merged**